layouts/shortcodes/product-grid.html

```diff
@@ -103,93 +332,150 @@
+        </div>
+        {{- end -}}
+    </fieldset>
+    </div>
+  </div>
+    <div class="filter-footer">
+      <button class="clear-button" onclick="resetFilters()">Clear filters</button>
+      <button class="apply-button" onclick="toggleVisibility('filter-popup')">Apply</button>
+    </div>
+  </div>
+</div>
+  <div class="ProductGrid--content" id="ProductGrid--content">
     {{- range $dataArray -}} 
-      {{- if and (eq $currentOption .group) (ne .show false) -}}
+      {{- if (ne .show false) -}}
         {{- $wrap := default false .wrap -}}
         <a
           class="ProductGrid--link"
           data-wrap-title="{{ $wrap }}"
           href="{{ .url }}"
+          data-product-group="{{ .group }}"
         >
           <span>{{ .title }}</span>
         </a>
         {{- end -}}
       {{- end -}}
+      
   </div>
+  <p id="emptyState" class="emptyState" style="display: none">Not finding what you need? Use our <a href="/search/">full search</a> or <a href="javascript:resetFilters();">reset the filters</a>.</p>
   </div>
-  {{- end -}}
-  <p id="emptyState" class="emptyState" style="display: none">Not finding what you need? Use our <a href="/search/">full search</a>.</p>
-</div>
-</div>
 
 <script>
-  function filterProducts() {
-    // Search input and items
-    var input = document.getElementById("search"); // User input
-    var query = input.value.toUpperCase(); // Make input non-case sensitive
-    var grid = document.getElementById("ProductGrid--range"); // Convert Hugo range to JS
-    var items = grid.getElementsByTagName("a"); // Extract tiles
-
-    // TODO: Search by URL, description (i.e. "cloudflare one" = Cloudflare Zero Trust)
-
-    // Iterate through products
-    for (var i = 0; i < items.length; i++) {
-      var item = items[i];
-      var text = item.innerText.toUpperCase();
-      if (text.indexOf(query) > -1) {
-        item.style.display = "";
-      } else {
-        item.style.display = "none";
-      }
+  document.addEventListener('DOMContentLoaded', function() {
+    // Listen for changes on all checkboxes
+    var checkboxes = document.getElementsByClassName("ProductCheckboxItem");
+    for (var i = 0; i < checkboxes.length; i++) {
+      checkboxes[i].addEventListener('change', filterList);
     }
+    
+    // Listen for input on the search box
+    var searchBox = document.getElementById('search');
+    searchBox.addEventListener('input', filterList);
+  });
 
-    var productAreas = document.getElementsByClassName("ProductGridSection")
-    var visibleSections = false;
+function filterList() {
+  var checkboxes = document.getElementsByClassName("ProductCheckboxItem");
+  var searchBox = document.getElementById('search');
+  // Get all checked checkboxes
+  var visibleItems = false;
+  var checkedValues = [];
+  for (var j = 0; j < checkboxes.length; j++) {
+    if (checkboxes[j].checked) {
+      checkedValues.push(checkboxes[j].id);
+    }
+  }
```

Comment:
```suggestion
  let numCheckboxes = 0;
  for (let j = 0; j < checkboxes.length; j++) {
    if (checkboxes[j].checked) {
      checkedValues.push(checkboxes[j].id);
      numCheckboxes++;
    }
  }
  let filterButton = document.getElementById('filter-button');
  if (numCheckboxes === 0) {
    filterButton.innerText = "Filter Products";
  } else {
    let suffix = numCheckboxes > 1 ? 's' : '';
    filterButton.innerText = `Filter Products (${numCheckboxes} group${suffix})`;
  }
```
I think that in the mobile version we should hint that there are filters being applied. I suggest that we update the filter button text to `Filter Results (N groups)` when we're filtering by one or more groups.